### PR TITLE
Enable use of CurvedCrackFront with crack_front_points

### LINKED ISSUE #6365: Enable use of CurvedCrackFront option with crack_front_points in DomainIntegral
If the crack front is defined with ```crack_front_points```, the ```CurvedCrackFront``` option for defining the direction of crack propagation with the ```crack_direction_method``` command doesn't currently work.

### PR DESCRIPTION
**Pull request by bwspenc**
closes #6365

### CODE REVIEW

**Comment by bwspenc** [comment]
@jiangwen84 Can you please review this?

File: modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C [hunk 1]
```diff
@@ -1058,7 +1051,7 @@ CrackFrontDefinition::rotateToCrackFrontCoords(const ColumnMajorMatrix tensor, c
 void
 CrackFrontDefinition::calculateRThetaToCrackFront(const Point qp, const unsigned int point_index, Real & r, Real & theta) const
 {
-  unsigned int num_points(_crack_front_points.size());
+  unsigned int num_points = getNumCrackFrontPoints();
   Point p = qp;
```
Comment (jiangwen84) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: Can we directly use ```const Point qp``` here?

**Comment by jiangwen84** [comment]
everything else looks great to me :+1:  :+1:  :+1:  :100:

File: modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C [hunk 2]
```diff
@@ -833,8 +826,8 @@ CrackFrontDefinition::updateDataForCrackDirection()
     _crack_plane_normal.zero();
 
     //Get 3 nodes on crack front
-    unsigned int num_nodes(_ordered_crack_front_nodes.size());
-    if (num_nodes<3)
+    unsigned int num_points = getNumCrackFrontPoints();
+    if (num_points<3)
```
Comment (permcody) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: operator spacing

File: modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C [hunk 3]
```diff
@@ -845,22 +838,22 @@ CrackFrontDefinition::updateDataForCrackDirection()
...
+      mid_id = (num_points-1)/3;
       end_id = 2*mid_id;
     }
     else
     {
       start_id = 0;
-      mid_id = (num_nodes-1)/2;
-      end_id = num_nodes-1;
+      mid_id = (num_points-1)/2;
+      end_id = num_points-1;
```
Comment (permcody) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: operator spacing here and just above

File: modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C [hunk 4]
```diff
@@ -845,22 +838,22 @@ CrackFrontDefinition::updateDataForCrackDirection()
...
-    Node & end   = _mesh.node(_ordered_crack_front_nodes[end_id]);
+    const Point *start = getCrackFrontPoint(start_id);
+    const Point *mid   = getCrackFrontPoint(mid_id);
+    const Point *end   = getCrackFrontPoint(end_id);
 
     //Create two vectors connecting them
-    RealVectorValue v1 = mid-start;
-    RealVectorValue v2 = end-mid;
+    RealVectorValue v1 = *mid - *start;
+    RealVectorValue v2 = *end - *mid;
```
Comment (permcody) [inline modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C]: Thanks!

**Comment by bwspenc** [comment]
@permcody I addressed the operator spacing issues, and with @jiangwen84's enthusiastic 3 thumbs up, I think this is ready to go.

As an aside, the Code Standards document doesn't explicitly spell out what the guidelines are for spacing around operators, so I'm always guessing on that.  The existing code base is not a good reference yet -- its style is all over the place in this regard. From what I see in the Code Standards, I'm gathering that there should be a space around all binary operators, but not around unary operators, and that there should be no spaces between parentheses and their contents.  Is that right?  Just a little bit of extra text there explicitly spelling out the requirements would be very helpful (as would be a check for this in the pre-test script and a script we could run to automatically fix infractions).

**Comment by permcody** [comment]
On Mon, Feb 22, 2016 at 3:09 PM Ben Spencer <notifications@github.com>
wrote:

> @permcody <https://github.com/permcody> I addressed the operator spacing
> issues, and with @jiangwen84 <https://github.com/jiangwen84>'s
> enthusiastic 3 thumbs up, I think this is ready to go.
>
> As an aside, the Code Standards document doesn't explicitly spell out what
> the guidelines are for spacing around operators, so I'm always guessing on
> that. The existing code base is not a good reference yet -- its style is
> all over the place in this regard. From what I see in the Code Standards,
> I'm gathering that there should be a space around all binary operators, but
> not around unary operators, and that there should be no spaces between
> parentheses and their contents. Is that right?
>
Yes, However the unary operators enforcement isn't as tight as the binary
operators. There are cases especially with tons of pointers and references
where it's nice to see space and sometimes less nice. We'll try to come to
an agreement on what we'd like but fixing the binary operators is a good
start.

**Comment by permcody** [comment]
:+1:

**Pull request merged**